### ISSUE TITLE
Web compatibility problem

### ISSUE THREAD

**Issue by Jack-Works** [frank-dspeed/tc39-proposal-importScripts#2]
Many codes use `typeof importScripts` to check if it is running in a Web Worker. Adding it globally breaks the ecosystem.

**Comment by frank-dspeed** [comment]
@Jack-Works i am aware of that but a research on http archive from google turned out that it is a problem that does not hit the 1% mark so it is do able also this is part of https://www.w3.org/community/wintercg/ so we address all engines that want to support web compat thats why i introduce it on this level i could also break the web or how you call it on a diffrent level if that is to rude 

but a feature check to guess a whole environment is a bit lazy anyway and needs to get deprecated anyway the question is only when and how. importScripts algins the whole ecosystem because it supplys a sync import as standard.

i would not even want to rename that as this clear tells what it does and it only does that. require for example handels resolve + 3 module systems + Scripts

**Comment by Jack-Works** [comment]
https://github.com/awesome-webextension/webpack-target-webextension/blob/master/lib/webpack5/RuntimeModules/LoadScript.js#L86

I don't think it is so uncommon, even code I wrote uses this pattern.

**Comment by frank-dspeed** [comment]
@Jack-Works i guess we can solve that on bundler and tooling level fast once it become a problem i do not see that as breaking but we will wait for more input i would be open to call it syncImportScripts() but all that makes less sence out of bigger view as most engines did implement already importScript or importScripts 

also calling it importScript and let it handle only a single arg would be a option but that is open to more discussion

**Comment by frank-dspeed** [comment]
## my final standpoint
if code like
```html
<script>var importScripts = () => {}</script>
```
or
script.js
```js
var importScripts = () => {}
```

```html
<script src="script.js"></script>
```

do break the whole web then the web is lost right?

**Comment by Jack-Works** [comment]
If you choose to write this, it's your setup. You break your website, and it's ok because you will fix it.

```html
<script>var importScripts = () => {}</script>
```

It's different when the _engine_ adds it. A previous example is `global` renamed to `globalThis`. https://github.com/tc39/proposal-global#naming

They also think "Further research has determined that using global will not break existing code.", but finally it does.

I can say naming it as `importScripts`, will 100% break _some_ website.

**Comment by frank-dspeed** [comment]
@Jack-Works i agree with you but thats true for Normativ proposals it is expected. but ok if enough people vote for it we go for 

```
importScript(url); // returns or throws
```

and it gets implemented ln pseudo code like: eval(load(url))
